### ISSUE TITLE
statfs path parameter missing

### ISSUE THREAD

**Issue by smessmer** [osxfuse/osxfuse#538]
Hello,

I noticed while developing https://www.cryfs.org that osxfuse doesn't pass through the `path` parameter to `statfs()` correctly. On Linux, it works as specified.

A minimal example:

- The filesystem is based on the hellofs example from here, but adds a statfs implementation that logs the path parameter it gets: https://pastebin.com/MhGmsLxZ
- Calling the file system is a small program that just calls `::statvfs()`: https://pastebin.com/QCSNi6sR

Compiling the file system and the program using it:

    $ clang++ -D_FILE_OFFSET_BITS=64 hellofs.cpp -losxfuse -ohellofs
    $ clang++ statfs.cpp -ostatfs

Running the file system:

    $ mkdir mountdir
    $ ./hellofs ./mountdir -f

Now, in another tab, call the program using the filesystem with the `hello.txt` file from the file system:

    $ ./statfs ./mountdir/hello.txt

And in the tab running the file system, we'll now see the output:

    Statfs with path: /

So basically, osxfuse didn't pass through the path parameter to the file system. On Linux, the output follows the specified behavior and looks like

    Statfs with path: /hello.txt

**Comment by bfleischer** [comment]
This is another case of macOS is not Linux. The macOS kernel does not pass the exact path you specified when calling statfs(2) to the FUSE kernel extension. I believe this makes more sense than the Linux approach. At least in case of macOS statfs(2) is supposed to return the same data for every path on a volume.
